devtools/src/org/labkey/devtools/ToolsController.java

```diff
@@ -365,4 +378,168 @@ public void addNavTrail(NavTree root)
             root.addChild("JSP Finder");
         }
     }
+
+    @RequiresPermission(AdminPermission.class)
+    public class CheckCrawlerActionsAction extends SimpleViewAction
+    {
+        @Override
+        public ModelAndView getView(Object o, BindException errors) throws IOException
+        {
+            String sourcePath = ModuleLoader.getInstance().getModule(DevtoolsModule.NAME).getSourcePath();
+
+            if (null == sourcePath)
+            {
+                errors.reject(ERROR_MSG, "Source path is null!");
+                return new SimpleErrorView(errors);
+            }
+
+            List<ControllerActionId> actionIds = new LinkedList<>();
+
+            // As of now, Crawler.java and the study tests are the only classes that specify crawler actions
+            for (String path : List.of(
+                sourcePath + "/../../../testAutomation/src/org/labkey/test/util/Crawler.java",
+                sourcePath + "/../study/test/src/org/labkey/test/tests/study")
+            )
+            {
+                File file = new File(path);
+                if (!file.exists())
+                {
+                    errors.reject(ERROR_MSG, FileUtil.getAbsoluteCaseSensitiveFile(file).getAbsolutePath() + ": path not found!");
+                    return new SimpleErrorView(errors);
+                }
+                addActionIds(actionIds, file);
+            }
+
+            List<ControllerActionId> missingModules = new ArrayList<>();
+            List<ControllerActionId> missingActions = new ArrayList<>();
```

Comment:
Make sets to avoid duplicate errors?
```suggestion
            Set<ControllerActionId> missingModules = new HashSet<>();
            Set<ControllerActionId> missingActions = new HashSet<>();
```

Reply:
Switched to TreeSet<>